--- a/cmdline/apt-key.in
+++ b/cmdline/apt-key.in
@@ -73,7 +73,7 @@
 	    unset TMPDIR
 	fi
     fi
-    GPGHOMEDIR="$(mktemp --directory --tmpdir 'apt-key-gpghome.XXXXXXXXXX')"
+    GPGHOMEDIR="$(mktemp -d 'apt-key-gpghome.XXXXXXXXXX')"
     CURRENTTRAP="${CURRENTTRAP} cleanup_gpg_home;"
     trap "${CURRENTTRAP}" 0 HUP INT QUIT ILL ABRT FPE SEGV PIPE TERM
     if [ -z "$GPGHOMEDIR" ]; then
@@ -318,30 +318,6 @@
 }
 
 is_supported_keyring() {
-    # empty files are always supported
-    if ! test -s "$1"; then
-	return 0
-    fi
-    local FILEEXT="${1##*.}"
-    if [ "$FILEEXT" = 'gpg' ]; then
-	# 0x98, 0x99 and 0xC6 via octal as hex isn't supported by dashs printf
-	if printf '\231' | cmp --silent --bytes=1 - "$1"; then
-	    true
-	elif printf '\230' | cmp --silent --bytes=1 - "$1"; then
-	    true
-	elif printf '\306' | cmp --silent --bytes=1 - "$1"; then
-	    true
-	else
-	    apt_warn "The key(s) in the keyring $1 are ignored as the file has an unsupported filetype."
-	    return 1
-	fi
-    elif [ "$FILEEXT" = 'asc' ]; then
-	true #dearmor_filename will deal with them
-    else
-	# most callers ignore unsupported extensions silently
-	apt_warn "The key(s) in the keyring $1 are ignored as the file has an unsupported filename extension."
-	return 1
-    fi
     return 0
 }
 
